hackathon_site/dashboard/frontend/src/slices/event/teamDetailSlice.ts

```diff
@@ -0,0 +1,98 @@
+import {
+    createAsyncThunk,
+    createEntityAdapter,
+    createSelector,
+    createSlice,
+} from "@reduxjs/toolkit";
+import { AppDispatch, RootState } from "slices/store";
+import { displaySnackbar } from "slices/ui/uiSlice";
+import { get } from "api/api";
+import { Team } from "api/types";
+
+const teamDetailAdapter = createEntityAdapter();
```

Comment:
are you sure that `Profile` is the correct type? when i use it i get this error when i attempt to populate the adapter:
![Screen Shot 2022-07-31 at 6 08 36 PM](https://user-images.githubusercontent.com/36282235/182047251-180eb0ae-c7c9-43d7-8a72-c25080a5b950.png)
i believe payload.profiles which i am feeding into the adapter is an array of  `ProfileWithoutTeamNumber & UserWithoutProfile` objects, but i don't know how to declare this union of types in the annotation

Reply:
Yes you're correct that was my mistake. If you look in types.ts lines 85-87 that is definition of the profile array type. This union is also used on lines 73-75 so we should have made a type for it a long time ago lol.
What you can do is create a new interface in types.ts as follows (place it on line 70):
```
export interface ProfileWithUser extends ProfileWithoutTeamNumber {
  user: UserWithoutProfile;
}
```
now you can use `ProfileWithUser` on lines 85 and 73 instead of what's there now and use the interface to type your adapter.